app/views/index/normal.phtml

```diff
@@ -75,41 +75,77 @@ $today = @strtotime('today');
 
 			$this->renderHelper('index/normal/entry_header');
 
-		?><div class="flux_content" dir="auto">
+		?><article class="flux_content" dir="auto">
 			<div class="content <?= $content_width ?>">
-				<h1 class="title"><a target="_blank" rel="noreferrer" class="go_website" href="<?= $this->entry->link() ?>"><?= $this->entry->title() ?></a></h1>
-				<div class="subtitle">
-					<div class="website"><a href="<?= _url('index', 'index', 'get', 'f_' . $this->feed->id()) ?>" title="<?= _t('gen.action.filter') ?>: <?= $this->feed->name() ?>">
+				<header>
+					<div class="website"><a href="<?= _url('index', 'index', 'get', 'f_' . $this->feed->id()) ?>" title="<?= _t('gen.action.filter') ?>">
 						<?php if (FreshRSS_Context::$user_conf->show_favicons): ?>
-							<img class="favicon" src="<?= $this->feed->favicon() ?>" alt="✇" loading="lazy" />
-						<?php endif; ?>
-						<span><?= $this->feed->name() ?></span></a>
+							<img class="favicon" src="<?= $this->feed->favicon() ?>" alt="✇" loading="lazy" /><?php
+						endif; ?><span><?= $this->feed->name() ?></span></a>
 					</div>
-					<div class="date"><?= $this->entry->date() ?></div>
-					<?php
+					<div class="tags">
+						<?php
+						$tags = FreshRSS_Context::$user_conf->bottomline_tags ? $this->entry->tags() : null;
+						if (!empty($tags)) {
+							?><?= _i('tag') ?><ul class="list-tags"><?php
+							$i = 0;
+							foreach ($tags as $tag) {
+								if ($i < 7) {
+									$i++;
+									?><li class="item tag"><a class="link-tag" href="<?= _url('index', 'index', 'search', '#' . str_replace(' ', '+', htmlspecialchars_decode($tag, ENT_QUOTES))) ?>" title="<?= _t('gen.action.filter') ?>">#<?= $tag ?></a></li><?php
+								} else { // more than 7 tags: show dropdown menu ?>
```

Comment:
The if/else should likely be outside the foreach

Reply:
why outside of the foreach? Within the foreach the `$i` is increased.

Reply:
I have coded what I meant: https://github.com/FreshRSS/FreshRSS/pull/4101/commits/f6239469748888cc07fa975cbc48792c60e50937
(Not tested) Please have a look.
It is not that is was not working, but more a matter of style: I did not like the fact that two logics were intertwined, requiring another variable to count, a break, and a foreach inside a foreach.

Reply:
`array_slice()` was the magic function that I did not know (and unfortunately did not search).
Thanks!